js/iframe-coordinator/src/FrameManager.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  HostToClient,
+  validate as validateOutgoing
+} from './messages/HostToClient';
+import {
+  ClientToHost,
+  validate as validateIncoming
+} from './messages/ClientToHost';
+
+const IFRAME_STYLE =
+  'position: absolute; top: 0; left: 0; width: 100%; height: 100%;';
+
+class FrameManager {
+  private _iframe: HTMLIFrameElement;
+  private _frameLocation: string;
+  private _window: Window;
+
+  constructor(mockWindow?: any) {
```

Comment:
Any reason to not use a configuration object instead of just a mock window parameter?

Reply:
It seemed like just passing the mock window for tests is simpler. I don't think it should introduce any comparability issues if the constructor starts taking arguments in the future, since users of the library shouldn't be passing anything in here. Maybe you're recommending a specific pattern I'm not familiar with though?

Reply:
I was just thinking if we have to pass 4-5 or more parameter s eventually.  Its better to have your options in a parameter object to reduce having to have loads of parameters to understand.